### PR TITLE
[CI] update oneDNN to v2.6

### PR DESCRIPTION
**Pull request by yangulei**
I submitted the PR to enable bfloat16 in DNNL BYOC(https://github.com/apache/tvm/pull/11111). But the PR failed to be built due to the version of the oneDNN in CI is limited to v2.2, which is a relative old one. So I update the CI script to get and build the latest oneDNN release.
I had tested the script on my own PC and it works fine, but I don't know how to test the script in CI environment. So let me know if I did something inappropriate, thanks!

cc @Mousius @areusch @driazati

### CODE REVIEW

File: docker/install/ubuntu_install_dnnl.sh [hunk 1]
```diff
@@ -20,10 +20,38 @@ set -e
...
+    else 
+        echo "downloading ${tar_file}."
+        tar_url="https://github.com/oneapi-src/oneDNN/archive/refs/tags/${tar_file}"
+        wget ${tar_url}
+    fi
+    tar -xzvf ${tar_file}
+fi
+
+cd ${src_dir}
+cmake . -DCMAKE_INSTALL_PREFIX=${install_dir} && make -j16 && make install
```
Comment (driazati) [inline docker/install/ubuntu_install_dnnl.sh:52]: nit: use all available CPUs
```suggestion
NPROC=$(nproc)
cmake . -GNinja -DCMAKE_INSTALL_PREFIX=${install_dir}
make -j"$NPROC"
make install
```

**Review by driazati — CHANGES_REQUESTED** [review]
CI tests your changes automatically so no worries there, I would prefer that we just upgrade the oneDNN version manually (and keep the version pinned) though instead of grabbing the latest (imagine that there is a breaking update, the Docker image build would just stop working one day)

**Comment by yangulei** [comment]
> CI tests your changes automatically so no worries there, I would prefer that we just upgrade the oneDNN version manually (and keep the version pinned) though instead of grabbing the latest (imagine that there is a breaking update, the Docker image build would just stop working one day)

Thanks for your comments and suggestions. I had modified the script to pin oneDNN to the latest release v2.6, and it's easy to change the pinned version or set to build the latest release.

**Comment by yangulei** [comment]
The CI failed due to connection error during the installation of tensorflow-aarch64 v2.6.2:

> [2022-04-28T01:43:42.770Z]` WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.VerifiedHTTPSConnection object at 0xffff827a3050>, 'Connection to snapshots.linaro.org timed out. (connect timeout=15)')': /ldcg/python-cache/tensorflow-aarch64/
> [2022-04-28T01:43:55.126Z] ERROR: Could not find a version that satisfies the requirement tensorflow-aarch64==2.6.2 (from versions: 1.2, 2.7.0rc0, 2.7.0rc1, 2.7.0, 2.7.1, 2.8.0rc0, 2.8.0rc1, 2.8.0, 2.9.0rc0, 2.9.0rc1)
> [2022-04-28T01:43:55.126Z] ERROR: No matching distribution found for tensorflow-aarch64==2.6.2
> [2022-04-28T01:44:00.406Z] WARNING: You are using pip version 19.3.1; however, version 22.0.4 is available.
> [2022-04-28T01:44:00.406Z] You should consider upgrading via the 'pip install --upgrade pip' command.
> [2022-04-28T01:44:00.406Z] The command '/bin/sh -c bash /install/ubuntu_install_tensorflow_aarch64.sh' returned a non-zero code: 1
> [2022-04-28T01:44:00.406Z] ERROR: docker build failed.
> script returned exit code 1

Should I do something or just ignore this?

**Comment by areusch** [comment]
@yangulei i think this means the CI's quota to PyPI was exceeded. perhaps retry now that it's been a while?

**Comment by driazati** [comment]
Yeah that index was down for a while last week (https://github.com/apache/tvm/pull/11156#issuecomment-1112015086) but it should be working now. If this happens again we might want to look into rehosting some of those assets

**Comment by yangulei** [comment]
@areusch @driazati 
PyPi works fine, thanks!
Looks like we should install the package to /usr/lib instead of /usr/local in the latest main branch. I have modified my script (follow the installation of gtest) and it pass all the CI now.

**Comment by yangulei** [comment]
Hi @areusch @driazati,
PR #11111 depends on this PR and had waited for 3 weeks, so I want this PR to be merged if it's ready.
Could you please take a look at this script and approve it if no more modifications are required?
Thanks a lot.

**Comment by yangulei** [comment]
Hi @areusch @driazati,
PR #11111 depends on this PR and had waited for 3 weeks, so I want this PR to be merged if it's ready.
Could you please take a look at this script and approve it if no more modifications are required?
Thanks a lot.

**Review by areusch — APPROVED** [review]
thanks @yangulei and sorry for the delay here!

**Comment by masahi** [comment]
@areusch We still need to go through the actual update process for the cpu image right? I've got another request for gpu update in https://github.com/apache/tvm/pull/11346, I can do the cpu update as well.

**Comment by yangulei** [comment]
Thank you @areusch, I will rebase #11111 once the docker image is ready.

**Pull request merged**